worker/worker.go

```diff
@@ -186,9 +186,23 @@ func (worker *Worker) createVersion(vcsManager *vcs.Manager, event *api.Event) {
 
 	destDir := vcsManager.GetCloneDir(&event.Service, &event.Version)
 	event.Data["context-dir"] = destDir
-	event.Data["image-name"] = fmt.Sprintf("%s/%s/%s", dockerManager.Registry,
-		strings.ToLower(event.Service.Username), strings.ToLower(event.Service.Name))
-	event.Data["tag-name"] = event.Version.Name
+
+	var imageName, tagName string
+
+	if "" == event.Service.ImageName {
+		imageName = fmt.Sprintf("%s/%s/%s", dockerManager.Registry,
+			strings.ToLower(event.Service.Username), strings.ToLower(event.Service.Name))
+		tagName = fmt.Sprintf("%d-%s", time.Now().Unix(), event.Version.Commit)
+	} else {
+		if arr := strings.Split(event.Service.ImageName, ":"); len(arr) == 2 {
+			imageName, tagName = arr[0], arr[1]
+		} else {
+			setEventFailStatus(event, err.Error())
```

Comment:
Here the `err` is nil, must new an error to illustrate the fail reason.

Reply:
done